core/server/i18n.js

```diff
@@ -69,27 +88,37 @@ I18n = {
      * @param {string} msgPath Path with in the JSON language file to desired string (ie: "errors.init.jsNotBuilt")
      * @returns {string}
      */
-    findString: function findString(msgPath, opts) {
+    findString: function findString(msgPath, defaultString, opts) {
         var options = _.merge({log: true}, opts || {}),
-            matchingString, path;
+            candidateString, matchingString, path;
 
         // no path? no string
         if (_.isEmpty(msgPath) || !_.isString(msgPath)) {
             chalk.yellow('i18n:t() - received an empty path.');
             return '';
         }
 
+        // If not in memory, load translations for core and theme.
         if (blos === undefined) {
             I18n.init();
+            I18n.loadThemeTranslations();
+        }
+
+        // Both jsonpath's dot-notation and bracket-notation start with '$'
+        // E.g.: $.store.book.title or $['store']['book']['title']
+        // The {{t}} translation helper passes here the full jsonpath with $
+        // Backend messages use dot-notation, and the $. is added here
+        if (msgPath.substring(0, 1) !== '$') {
+            path = '$.' + msgPath;
+        } else {
+            path = msgPath;
         }
 
-        matchingString = blos;
+        // jp.value is a jsonpath method. Info:
+        // https://www.npmjs.com/package/jsonpath
+        candidateString = jp.value(blos, path) || defaultString;
```

Comment:
If i reset my db and then serve to /ghost, i am getting

> MESSAGE: Lexical error on line 1. Unrecognized text.
$.errors.update-check.checkingForUp
---------------^
Error: Lexical error on line 1. Unrecognized text.
$.errors.update-check.checkingForUp

Reply:
I assume it does not support `-` keys, which it should.

Reply:
That seemed to be one of the side effects of appending the theme strings to the core strings as one monolithic `blos` object, applying one system to two different situations.

I think this is probably one of the multiple issues fixed by today's commit _[i18n] Separate `coreStrings` and `themeStrings` objects_.

Reply:
The problem with the hyphen `-` is that, while json's _bracket-notation_ (e.g. `$['store']['book']['title']`) allows any international Unicode characters such as Japanese, etc. in keys, Ghost's backend messages currently use _dot-notation_ for paths (e.g. `store.book.title`, which is `$.store.book.title` in json's _dot-notation_), where only word characters (that is `\w` or `[A-Za-z0-9_]` in RegExp) are allowed for identifier names. That's in dchester's [jsonpath](https://www.npmjs.com/package/jsonpath) implementation already included with Ghost, which I think is the safest one, and also the first in npm's search results.

This only affects to one identifier in the current Ghost master's _core/server/translations/en.json_ file. It's `update-check`, the only one with hyphen, which is called by _core/server/update-check.js_ with for example `common.i18n.t('errors.update-check.checkingForUpdatesFailed.error')`.

Two possible solutions, keeping the current simple nested _dot-notation_ for backend messages:

1. For now, when separating core and theme strings, while naturally using jsonpath for themes, I've kept instead the current Ghost master's method in _i18n.js_ for core backend messages, which is a non-standard but practical one, that just separates by dots, therefore allowing hyphens, etc.

2. Another probably better way would be to use standard jsonpath for both core and themes, but without that hyphen (currently only that one in `en.json`) when using _dot-notation_. That is, changing just `update-check` to `updateCheck`. The rest of keys in core _en.json_ already use camelCase without hyphens. This would use `common.i18n.t('errors.updateCheck.checkingForUpdatesFailed.error')`, etc. in the file _core/server/update-check.js_ .

You can say me if it's ok to go with the 2nd option. I think it's the best and simplest one.